attrs_strict/_type_validation.py

```diff
@@ -51,6 +72,25 @@ def _validate_elements(attribute, value, expected_type):
         _handle_tuple(attribute, value, expected_type)
     elif base_type == typing.Union:
         _handle_union(attribute, value, expected_type)
+    elif base_type in {typing.Callable, Callable}:
+        _handle_callable(attribute, value, expected_type)
+
+
+def _handle_callable(attribute, callable_, expected_type):
+    _signature = signature(callable_)
+    callable_args = [
+        param.annotation for param in _signature.parameters.values()
+    ]
+    callable_args.append(_signature.return_annotation)
+    expected_args = expected_type.__args__
+    if not expected_args:
+        return  # No annotations specified on type, matches all Callables
+
+    for callable_param, expected_param in zip_longest(
+        callable_args, expected_args
+    ):
+        if callable_param != expected_param:
```

Comment:
Would this work for cases like `String` and `str` ? How about `Union` types or nested types like `Dict[str, int]` and `Dict` ? I'm not convinced this is enough ...

Reply:
Good question, will look into this

Reply:
We only need to handle `Union`, the rest of the types can be compared directly. eg `typing.Dict[str, int] == typing.Dict[str, int]`

We might need to have a special case for types like `AnyStr` but those might come under `TypeVars` 🤔

Reply:
One case that fails, is the case for nested `Unions` or something like 
```python
typing.Callable[[typing.Union[int, typing.Dict[typing.Union[str, int], str]]], int]
```

Looking into this 🤔